src/Vault.sol

```diff
@@ -0,0 +1,261 @@
+// SPDX-License-Identifier: BSD Clause-3
+pragma solidity ^0.8.24;
+
+import {
+    AccessControlUpgradeable,
+    Address,
+    ERC20PermitUpgradeable,
+    IERC20,
+    Math,
+    ReentrancyGuardUpgradeable,
+    SafeERC20
+} from "./Common.sol";
+
+import {Storage} from "src/Storage.sol";
+import {IVault} from "src/interface/IVault.sol";
+
+contract Vault is IVault, ERC20PermitUpgradeable, AccessControlUpgradeable, ReentrancyGuardUpgradeable {
```

Comment:
What's the intention on how to handle allocation execution?

Example:

100 ETH is deposited.

Vault

deposits  10 ETH to ynETH
converts 50 ETH to stETH and allocates it to ynLSDE
deposits 40 ETH to A Third party ERC4626 strategy

Reply:
Suppose that there are 2 strategies that can take stETH

ynLSDe (eigenlayer)
ynLSDk (karak)

How can the vault decide to distribute assets here ?

Reply:
Also each strategy has potentially different deposit signature calls

Eg.

deposit(address. asset, uint256 amount, address receiver)
deposit(uint256 amount, address receiver)
or some other version.

In the beginning the `bytes calldata` parameter was considered to inject custom calls with some guard rails around those.

We'd have to have a way of doing that

Reply:
As discussed in the call, sounds like we'd want

the strategy controller/governor  to be able to call a limited set of transactions  with whitelisted targets

Reply:
> Suppose that there are 2 strategies that can take stETH
> 
> ynLSDe (eigenlayer) ynLSDk (karak)
> 
> How can the vault decide to distribute stETH assets here ?
> 
> Eg: 100 stETH enters the vault, and those 2 stETH strategies exist

Perhaps we can start with a permissioned Governor function that is able to deposit or withdraw from strategy vaults.

Reply:
> Also each strategy has potentially different deposit signature calls
> 
> Eg.
> 
> deposit(address. asset, uint256 amount, address receiver) deposit(uint256 amount, address receiver) or some other version.
> 
> In the beginning the `bytes calldata` parameter was considered to inject custom calls with some guard rails around those.
> 
> We'd have to have a way of doing that

Yeah, I guess we can whitelist to only call strategy contracts as targets, and will have to experiment with how to make the func signatures safe, but generalized.

Reply:
> As discussed in the call, sounds like we'd want
> 
> the strategy controller/governor to be able to call a limited set of transactions with whitelisted targets

How about:
Governor sets the strategies:
operator can call any function on the strategies with calldata, but txes are restricted to only calling the strategy contracts.

Reply:
> Also each strategy has potentially different deposit signature calls
> 
> Eg.
> 
> deposit(address. asset, uint256 amount, address receiver) deposit(uint256 amount, address receiver) or some other version.
> 
> In the beginning the `bytes calldata` parameter was considered to inject custom calls with some guard rails around those.
> 
> We'd have to have a way of doing that

yes, I'll take a look at how to implement this and ping you with it today or tomorrow

Reply:
Picking up these questions in the scenario tests folder